### PR TITLE
Remove system partition

### LINKED ISSUE #4166: Investigate usage of system group
**Is your feature request related to a problem? Please describe.**

I don't think it's 100% clear to use what the usage of the system partition is. To some extent it is used to centralize which primitives exist, how they are distributed, etc., but it could be more. I would like to know and document what its use it, to know if in the long run we actually need it.

**Describe the solution you'd like**

A clear documentation of what the system partition is used for, and if it is necessary for our use case, i.e. if we could get rid of it in the long run.

**Describe alternatives you've considered**

Do nothing :+1: 

**Additional context**

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

Removes the system partition from our code base, since it was no longer used.

Run a normal benchmark and saw no related issues. I would expect some on start up, if there are any.

sys

I discussed with @deepthidevaki to remove the Bootstrap test, since it is not really deterministic / guaranteed whether the nodes become really leader or not, this means the test was really flaky.

## Related issues

closes #4166 

## Pull Request Checklist

- [X] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [X] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [X] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

**Review by deepthidevaki — APPROVED** [review]
Looks good to me. :+1:

File: atomix/cluster/src/main/java/io/atomix/primitive/partition/PartitionManagementService.java [hunk 1]
```diff
@@ -44,18 +43,4 @@
    * @return the primitive type registry
    */
   PrimitiveTypeRegistry getPrimitiveTypes();
-
-  /**
-   * Returns the primary election service.
-   *
-   * @return the primary election service
-   */
-  PrimaryElectionService getElectionService();
```
[the remark this answers is not in the document]
Comment (Zelldon) [reply atomix/cluster/src/main/java/io/atomix/primitive/partition/PartitionManagementService.java:53]: @deepthidevaki I think you're right. I removed more classes see below

**Comment by Zelldon** [comment]
I'm not sure whether I can merge this PR the tests arefailing quite often now

**Comment by Zelldon** [comment]
Bors r+

**Pull request merged**